### PR TITLE
Feature

### LINKED ISSUE #7: Rate app dialog
Implement Rate app dialog from [Hash Checker](https://github.com/hash-checker/hash-checker).

### PR DESCRIPTION
**Pull request by vipozdn**
## Checklist

__Common__

- [x] I am ran the app before creating PR
- [x] I am ran all tests before creating PR

__UI__

- [x] I am ran the app for visual checks.

__Logic__

- [x] I am tested basic app functionality.

## Changes

Describe all changes here:

- First;
- Second;
- Third;
- ...

## Comments

Describe all additional information here.

### CODE REVIEW

File: app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/hashcalculator/ui/HashCalculatorFragment.java [hunk 1]
```diff
@@ -97,6 +100,24 @@
             );
         } else {
             etGeneratedHash.setText(hashValue);
+            SharedPreferencesSettingsHelper settingsHelper = new SharedPreferencesSettingsHelper(context);
```
Comment (fartem) [inline app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/hashcalculator/ui/HashCalculatorFragment.java]: You need to use `SettingsHelper` instance from Dagger injection. If a class don't contains `SettingsHelper` injection, add it as local public field with `@Inject` annotation.

**Review by fartem — CHANGES_REQUESTED** [review]
Update `SettingsHelper` injection in `HashCalculatorFragment`.

**Comment by fartem** [comment]
@vipozdn Also, you can copy instrumentation tests for Rate dialog from [here](https://github.com/hash-checker/hash-checker/blob/master/app/src/androidTest/java/com/smlnskgmail/jaman/hashchecker/rateapp/RateAppDialogTest.java).

File: app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/settings/api/SettingsHelper.java [hunk 2]
```diff
@@ -41,4 +43,7 @@
 
     void setRefreshSelectedFileStatus(boolean status);
 
+    boolean canShowRateAppDialog(Context context);
```
Comment (fartem) [inline app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/settings/api/SettingsHelper.java]: Delete `Context` argument from `canShowRateAppDialog` and `increaseHashGenerationCount` methods. Context available in interface realization (`SharedPreferencesSettingsHelper`).

**Review by fartem — CHANGES_REQUESTED** [review]

**Comment by fartem** [comment]
@vipozdn Also, check [Codacy report](https://app.codacy.com/gh/hash-checker/hash-checker-lite/pullRequest?prid=7422198). In last commit you have some code issues.

File: app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/settings/api/SettingsHelper.java (hunk 2 above)
Comment (vipozdn) [reply app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/settings/api/SettingsHelper.java]: okay!

**Review by fartem — APPROVED** [review]

**Comment by fartem** [comment]
@vipozdn thanks.

**Comment by vipozdn** [comment]
> 
> 
> @vipozdn thanks.

I'm always happy to help!

**Comment by fartem** [comment]
@vipozdn Also, your PR contains some issues from [CPD](https://pmd.github.io/latest/pmd_userdocs_cpd.html). You can fix issues and assign [this](https://github.com/hash-checker/hash-checker-lite/issues/70) issue. CPD checks run with `cpdCheck` task by Gradle.

**Comment by vipozdn** [comment]
> @vipozdn Also, your PR contains some issues from [CPD](https://pmd.github.io/latest/pmd_userdocs_cpd.html). You can fix issues and assign [this](https://github.com/hash-checker/hash-checker-lite/issues/70) issue. CPD checks run with `cpdCheck` task by Gradle.

Okay

**Pull request merged**